libfwupd/fwupd-client-sync.c

```diff
@@ -958,6 +958,7 @@ fwupd_client_get_bios_attrs_cb(GObject *source, GAsyncResult *res, gpointer user
 /**
  * fwupd_client_get_bios_attrs:
  * @self: a #FwupdClient
+ * @authenticate: Whether to use PolicyKit to authorize
```

Comment:
this seems like the tail wagging the dog. Can't we use the interactive hint from the client?

Reply:
Not if we want bash completion to work. That's an interactive terminal usage.

Reply:
Ahh, but not interactive as in "vt active" more a fwupd client hint.

Reply:
So like a feature flag to indicate whether to use PK for a command?

Reply:
Is that sane? It *feels* per-client rather than per-request.

Reply:
It's not per client though. You will want to have the ability to do it for fwupdmgr both ways. I also think in the GUI tools you would need to have the ability to get some information before you authenticate.

None of the gnome panels require you to unlock the page just to view it. It's only once you want to interact with the page you authenticate.

Reply:
> You will want to have the ability to do it for fwupdmgr both ways

Ya, maybe. I do think we're possibly painting ourselves into a corner with just a boolean tho. Maybe a `FwupdClientGetBiosAttrsFlags` guint64?

Reply:
How would you feel about the client calling set feature flags multiple times?  Then thinking through the GUI case there can be a set feature flags call that says `FWUPD_FEATURE_FLAG_CAN_AUTHENTICATE` (or if you can think of a better name).  And then we can drop having an argument all together on this.  GUI (and bash completion case) can avoid setting that.

Reply:
> the client calling set feature flags multiple times

I think that works fine now, no?

> FWUPD_FEATURE_FLAG_CAN_AUTHENTICATE

I think it needs to be the other way around for compat; old gnome-software won't be setting that flag, and if any of the other methods get the same kind of filtering in the future then we don't want to break API. e.g. FWUPD_FEATURE_FLAG_NO_AUTHENTICATE

Reply:
Maybe that's not true; maybe we can fix it for old clients either-way.

Reply:
OK, take a look at what I've done now.  I agree this generally scales better.